cmd/zc_enumerator.go

```diff
@@ -593,44 +597,136 @@ type syncEnumerator struct {
 	secondaryTraverser ResourceTraverser
 
 	// the results from the primary traverser would be stored here
-	objectIndexer *objectIndexer
+	objectIndexer *folderObjectIndexer
 
 	// general filters apply to both the primary and secondary traverser
 	filters []ObjectFilter
 
 	// the processor that apply only to the secondary traverser
 	// it processes objects as scanning happens
 	// based on the data from the primary traverser stored in the objectIndexer
-	objectComparator objectProcessor
+	objectComparator folderProcessor
```

Comment:
What's the rationale behind introducing folderProcessor? That seems to be adding unnecessary complexity.
Is it because you are suspecting golang will not GC memory for deleted map entries unless the entire map is deleted?